diff --git a/at_core/inc/plugins.inc b/at_core/inc/plugins.inc
--- a/at_core/inc/plugins.inc
+++ b/at_core/inc/plugins.inc
@@ -17,7 +17,7 @@
  * @param $theme_name, usually the active theme.
  */
 function at_get_plugins($theme_name) {
-  $plugins = drupal_static(__FUNCTION__, array());
+  $plugins = &drupal_static(__FUNCTION__, array());
   if (empty($plugins)) {
     if ($plugins_cache = cache_get('at_get_plugins_plugins')) {
       $plugins = $plugins_cache->data;
